src/main/java/Date.java

```diff
@@ -0,0 +1,42 @@
+import java.text.SimpleDateFormat;
+
+/**
+ * Date representation of date string that's given
+ */
+public class Date {
+    boolean exists = false;
+    String precursor = "";
+    SimpleDateFormat formatter = new SimpleDateFormat("dd/MM/yy HHss");
+    java.util.Date data;
```

Comment:
Putting the date import at the start of the file as per convention would be better practice!